scripts/hatvaultsdeploy.js

```diff
@@ -1,77 +1,84 @@
 const ADDRESSES = require("./addresses.js");
-async function main() {
+async function main(
+  rewardsToken = "0x51a6Efc15c50EcE1DaAD1Ee4fbF8DEC76584c365",
+  rewardPerBlock = "16185644800000000",
+  startBlock = null,
+  multiplierPeriod = "195200",
+  governance = ADDRESSES[network.name].governance,
+  swapToken = "0x51a6Efc15c50EcE1DaAD1Ee4fbF8DEC76584c365",
+  whitelistedRouters = ["0xE592427A0AEce92De3Edee1F18E0157C05861564"],
+  tokenLockFactory = "0x6E6578bC77984A1eF3469af009cFEC5529aEF9F3",
+  silent = false
+) {
   // This is just a convenience check
-  if (network.name === "hardhat") {
+  if (network.name === "hardhat" && !silent) {
     console.warn(
       "You are trying to deploy a contract to the Hardhat Network, which" +
         "gets automatically created and destroyed every time. Use the Hardhat" +
         " option '--network localhost'"
     );
   }
 
-  const addresses = ADDRESSES[network.name];
   // ethers is avaialble in the global scope
   const [deployer] = await ethers.getSigners();
   const deployerAddress = await deployer.getAddress();
 
-  //constructor params for test
-  //rinkeby hat
-  const rewardsToken = "0x51a6Efc15c50EcE1DaAD1Ee4fbF8DEC76584c365";
-  const rewardPerBlock = "16185644800000000";
-  const startBlock = await ethers.provider.getBlockNumber();
-  const multiplierPeriod = "195200";
-  let governance = addresses.governance;
+  if (startBlock === null) {
+    startBlock = await ethers.provider.getBlockNumber();
+  }
   if (!governance && network.name === "hardhat") {
     governance = deployerAddress;
   }
 
-  //const governance  = await deployer.getAddress();
-  //v3 router
-  const uniSwapRouter = "0xE592427A0AEce92De3Edee1F18E0157C05861564";
-  const tokenLockFactory = "0x6E6578bC77984A1eF3469af009cFEC5529aEF9F3";
-
-  console.log("Deploying the contracts with the account:", deployerAddress);
-  console.log("Account balance:", (await deployer.getBalance()).toString());
+  if (!silent) {
+    console.log("Deploying the contracts with the account:", deployerAddress);
+    console.log("Account balance:", (await deployer.getBalance()).toString());
+  }
 
   const HATVaults = await ethers.getContractFactory("HATVaults");
-  const hatVaults = await HATVaults.deploy(
+  const hatVaults = await upgrades.deployProxy(HATVaults, [
```

Comment:
we should also decide on which proxy pattern to use: transparent or uups

It's a setting: https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades